dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/client/DefaultServiceInstance.java

```diff
@@ -77,14 +77,12 @@ public DefaultServiceInstance(DefaultServiceInstance other) {
     }
 
     public DefaultServiceInstance(String serviceName, String host, Integer port) {
-        if (port != null && port.intValue() < 1) {
+        if (port != null && port < 1) {
             throw new IllegalArgumentException("The port must be greater than zero!");
         }
         this.serviceName = serviceName;
         this.host = host;
         this.port = port;
```

Comment:
(this.port = port) it will cause a npe when port == null, I'm not sure it's ok
why not add a null check
```
if (port == null) {
    throw new IllegalArgumentException("The port should not be null!");
}
if (port < 1) {
    throw new IllegalArgumentException("The port must be greater than zero!");
}
```

Reply:
yep, add null checker will be better.